src/Features/Core/Portable/Shared/Extensions/DocumentExtensions.cs

```diff
@@ -147,5 +146,29 @@ public static async Task<NamingRule> GetApplicableNamingRuleAsync(this Document
 
             throw ExceptionUtilities.Unreachable();
         }
+
+        public static async Task<ImmutableArray<DiagnosticData>> GetCachedCopilotDiagnosticsAsync(this TextDocument document, TextSpan span, CancellationToken cancellationToken)
```

Comment:
this should not be an extension on TextDocument.  Also, if it is, it should be in an extensions namespace that brings in copilot.